pyccel/parser/semantic.py

```diff
@@ -2178,7 +1913,7 @@ def _visit_Assign(self, expr, **settings):
 
                 # TODO imporve this will not work for
                 # the case of different results types
```

Comment:
Is this comment still accurate?

Reply:
yes

Reply:
If we have multiple results with different types, shouldn't `rhs.dtype` be a PythonTuple?

Reply:
yes it should be a  PythonTuple but I'm not sure yet on how to make it working

Reply:
@EmilyBourne Any ideas on this point?

Reply:
Not without more details on why it doesn't work. I would have expected the only change necessary would be to add Interface to the types here: https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L2229

Reply:
We'll improve this in the future.